### PR TITLE
[cxxmodules] Re-introduce Cling modules

### PR DESCRIPTION
**Pull request by yamaguchi1024**
This is a revert (316f553538d2474c566e0b5460ce79d90e3028c0) of revert
 (76a9ce75d2c29c169812b0816163ac7ce27ffad5 and
88c16588108508eff66cc160d244c4a59eb157ed).

### CODE REVIEW

File: cmake/modules/RootNewMacros.cmake [hunk 1]
```diff
@@ -293,7 +293,6 @@ function(ROOT_GENERATE_DICTIONARY dictionary)
 
   if(CMAKE_PROJECT_NAME STREQUAL ROOT)
     set(includedirs -I${CMAKE_SOURCE_DIR}
-                    -I${CMAKE_SOURCE_DIR}/interpreter/cling/include # This is for the RuntimeUniverse
```
Comment (vgvassilev) [inline cmake/modules/RootNewMacros.cmake]: This will likely not build a module for RuntimUniverse.h and friends.

**Comment by yamaguchi1024** [comment]
Modules is broken now, and we're still working on this PR because we cannot reproduce the error locally.

**Comment by vgvassilev** [comment]
You can harvest all relevant include files from the remote machine and copy them over to a local machine. Then you can readjust the rootcling invocation and start reducing. I believe that’s the way to go.

**Comment by yamaguchi1024** [comment]
@phsft-bot build just on slc6/gcc62 with flags -Druntime_cxxmodules=On -Dctest_test_exclude_none=on

**Comment by yamaguchi1024** [comment]
@phsft-bot build just on slc6/gcc62 with flags -Druntime_cxxmodules=On -Dctest_test_exclude_none=on

**Comment by yamaguchi1024** [comment]
@phsft-bot build just on slc6/gcc62 with flags -Druntime_cxxmodules=On -Dctest_test_exclude_none=on

**Comment by vgvassilev** [comment]
@phsft-bot build just on slc6/gcc62 with flags -Dccache=ON -Druntime_cxxmodules=On

**Comment by yamaguchi1024** [comment]
@phsft-bot build just on slc6/gcc62 with flags -Druntime_cxxmodules=On -Dctest_test_exclude_none=on

**Comment by yamaguchi1024** [comment]
@phsft-bot build

**Comment by yamaguchi1024** [comment]
@phsft-bot build

**Comment by yamaguchi1024** [comment]
@phsft-bot build

**Pull request merged**